### DISCUSSION TITLE
Map l key to open files and traverse into directories

### DISCUSSION THREAD

**Discussion by fritzrehde** [sxyazi/yazi#494] [1. Q&A] [answered]
Coming from lf and ranger, I am used to the behaviour where pressing the l key causes files to be opened *and* directories to be traversed into. In yazi, it seems like the default is that l is mapped to traversing into directories and o is for opening files. Is there a way to do both, conditionally on whether operating on a file or directory, with one command/key binding?

**Comment by sxyazi** [comment, marked as answer]
See https://yazi-rs.github.io/docs/faq#why-cant-open-and-enter-be-a-single-command

And use this tip (main branch needed for now), https://yazi-rs.github.io/docs/tips#smart-enter-enter-for-directory-open-for-file

**Reply by fritzrehde** [reply]
Thanks, sorry for not reading the documentation properly before asking my question, the feature as well as the reason it exists (i.e. why `open` and `enter` are separate) are very well-explained there.
